cub/test/test_nvtx_disabled.cu

```diff
@@ -0,0 +1,15 @@
+#define CUB_DETAIL_BEFORE_NVTX_RANGE_SCOPE(name) static_assert(false, "");
+#define CCCL_DISABLE_NVTX
+
+#include <cub/device/device_for.cuh>
+
+#include <thrust/iterator/counting_iterator.h>
+
+#include <cuda/std/functional>
+
```

Comment:
TODO: Maybe add this suggestion from @gevtushenko:
```
#ifdef CCCL_DISABLE_NVTX
#ifdef NVTX_VERSION
#error "Should not include NVTX when CCCL_DISABLE_NVTX is defined"
#endif
#endif
```
To make sure we are also not including the NVTX headers by accident.

Reply:
Added.